obal/data/modules/srpm.py

```diff
@@ -0,0 +1,89 @@
+"""
+Build SRPM
+"""
+
+import shutil
+import os
+import commands
+from tempfile import mkdtemp
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.chdir import chdir
+
+def run_command(command):
+    return commands.getstatusoutput(command)
+
+def copy_sources(spec_file, package_dir, sources_dir):
+    command = "spectool --list-files %s" % spec_file
+    _status, sources = run_command(command)
+
+    source_files = []
+
+    for source in sources.split("\n"):
+        if 'http' not in source:
+            shutil.copy("%s/%s" % (package_dir, source.split(' ')[1]), sources_dir)
+        else:
+            with chdir(package_dir):
+                run_command("git-annex lock")
+                _status, annexed_files = run_command("git-annex find --include='*'")
+                annexed_files = annexed_files.splitlines()
+                run_command("git-annex get")
+                run_command("git-annex unlock")
+
+                for annex in annexed_files:
+                    shutil.copy(annex, sources_dir)
+
+                run_command("git-annex lock --force")
+
+
+def main():
+    """
+    Build a package using tito
+    """
+    module = AnsibleModule(
+        argument_spec=dict(
+            package=dict(type='str', required=False),
+            output=dict(type='path', required=False),
+        )
+    )
+
+    package = module.params['package']
+    output = module.params['output']
+
+    spec_file = '%s/%s.spec' % (package, os.path.basename(package))
+
+    base_dir = mkdtemp()
+    sources_dir = os.path.join(base_dir, 'SOURCES')
+    build_dir = os.path.join(base_dir, 'BUILD')
+
+    os.mkdir(sources_dir)
+    os.mkdir(build_dir)
+
+    copy_sources(spec_file, package, sources_dir)
+    shutil.copy(spec_file, base_dir)
+
+    command = ['rpmbuild', '-bs']
+    command.append('--define "_topdir %s"' % base_dir)
+    command.append('--define "_sourcedir %s"' % sources_dir)
+    command.append('--define "_builddir %s"' % build_dir)
+    command.append('--define "_srcrpmdir %s"' % base_dir)
+    command.append('--define "_rpmdir %s"' % base_dir)
```

Comment:
From my previous research into this, this doesn't seem to be the _complete_ list of --defines we'll need.

One particular one I can think of is dist, to ensure we generate the correct one for the correct location.

Another is the scl macros.  This _may_ not be too big of a deal with koji's newfound ability to rebuild SRPMS, but i think it's worth noting for completeness that there are a few defines wie'll need.

Reply:
[Link to tito's dist defines](https://github.com/dgoodwin/tito/blob/master/src/tito/builder/main.py#L211-L220)

Reply:
To support dist, I think we'd have to start defining the dists as part of the package manifest metadata and we would have to require that as part of the command requirements. I could add optional support for declaring dist? So that it you can do:

```
obal srpm foo
obal srpm foo --dist=.el7
```

Reply:
Good point. Currently we also set `foremandist` via tito.props.

Reply:
I'd say... We will need to define releasers globally somehow, with specific ability to define the rpm macros we need per releaser that we specify today in tito.props

Reply:
I'm fairly certain the the local dist does not affect the built RPMs dist. As this is controlled by macros inside the buildroot.